### PR TITLE
snapcraft: don't manage loopback device and add patch for more relaxed ipv6 method matching

### PR DESCRIPTION
**Pull request by Meulengracht**
Support for managing the loopback device by default was introduced in 1.42 (i.e in 24 version of NM) - https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/merge_requests/1332

However, this is causing connection loss on reinstall of network-manager. This is most likely due to a possible bug in the interaction between netplan and NM for the loopback device.

It seems upon initial install that the loopback device is only managed externally (i.e it's not migrated directly to NM by default config). 
* If we then run `nmcli con up lo` it will become fully managed by NM
* If not, then it stays externally managed (i.e just tracked by NM)

In the case of being externally tracked by NM, this means that once netplan apply runs again, that netplan thinks it's managed by NM, when in actually it's only tracked.

So when NM is reinstalled, it disconnects it and flushes it's IP addresses, causing the loopback to reset, and NM fails to bring up the connection again (since the external device has changed, it assumes something else is managing it), and sees the device as completely unmanaged.

This reverts the behaviour to the pre-1.42 behavior.

Once network-manager is in stable I will revert the `24/edge` commit

### CODE REVIEW

File: tests/core/no-netplan-default-config/task.yaml [hunk 1]
```diff
@@ -34,10 +34,8 @@ execute: |
     networkctl status $eth_if | MATCH '(unmanaged|failed)'
     network-manager.nmcli c show --active | MATCH $eth_if | MATCH -v netplan
 
-    # Verify the active connections
-    test `network-manager.nmcli -m multiline -f UUID c show --active | wc -l` -eq 2
-    network-manager.nmcli -m multiline -f DEVICE c show --active | MATCH "lo"
-    network-manager.nmcli -m multiline -f DEVICE c show --active | MATCH "ens4"
+    # We should only have a single active configuration
+    test `network-manager.nmcli -m multiline -f UUID c show --active | wc -l` -eq 1
```
Comment (valentindavid) [inline tests/core/no-netplan-default-config/task.yaml]: I am surprised we still use backtick for command substitution.

Comment (Meulengracht) [reply tests/core/no-netplan-default-config/task.yaml]: This was indeed a revert of an older commit I had done, but I can change this to subshell

File: snap-common/startup-hooks/10-globally-managed-devices.sh
Comment (alfonsosanchezbeato) [inline snap-common/startup-hooks/10-globally-managed-devices.sh:1]: Cannot be this simply added to `/etc/NetworkManager/NetworkManager.conf`? I see we use a hook also to create `enable-wol.conf`, but actually that could go to the static configuration as well, not sure why that was not done originally like that.

File: snap-common/etc/NetworkManager/NetworkManager.conf [hunk 2]
```diff
@@ -10,4 +10,5 @@ dhcp=internal
 managed=false
 
 [device]
+keep-configuration=no
```
Comment (alfonsosanchezbeato) [inline snap-common/etc/NetworkManager/NetworkManager.conf]: Is this enough to solve the issue or do we also need the fix for the loopback device? I do not see the relantionship, although in any case I'm happy also with not taking control of the loopback, but I'm curious.

File: tests/core/no-netplan-default-config/task.yaml (hunk 1 above)
Comment (alfonsosanchezbeato) [inline tests/core/no-netplan-default-config/task.yaml]: The check for ens4 should still happen I guess

**Review by alfonsosanchezbeato — COMMENTED** [review]
Thanks for digging into this! Some comments below.

File: snap-common/etc/NetworkManager/NetworkManager.conf (hunk 2 above)
Comment (Meulengracht) [reply snap-common/etc/NetworkManager/NetworkManager.conf]: Based on your knowledge sharing last daily I dug into NM's source code and figured out why it was not matching the connections. I have patched NM to allow matching based on my findings, and removed this config change

File: snap-common/startup-hooks/10-globally-managed-devices.sh
Comment (Meulengracht) [reply snap-common/startup-hooks/10-globally-managed-devices.sh:1]: In this specific case, the naming of the config-file actually means what it says. `10-globally-managed-devices` is an unique identifier that NM treats exactly like it says. So we must create the file AFAICT

File: tests/core/no-netplan-default-config/task.yaml (hunk 1 above)
Comment (Meulengracht) [reply tests/core/no-netplan-default-config/task.yaml]: I'll do these test improvements in a followup

File: snap-patch/networkmanager/0003-src-core-allow-reverse-ipv6-disabled-settings.patch
Comment (alfonsosanchezbeato) [inline snap-patch/networkmanager/0003-src-core-allow-reverse-ipv6-disabled-settings.patch:1]: I think this patch is quite reasonable and maybe upstream is willing to accept it.

**Review by alfonsosanchezbeato — APPROVED** [review]
LGTM, thank you!

**Pull request merged**